pkg/asset/installconfig/gcp/dns.go

```diff
@@ -29,6 +31,21 @@ func GetPublicZone(ctx context.Context, project, baseDomain string) (*dns.Manage
 	return dnsZone, nil
 }
 
+// ParseZoneName will attempt to extract the project and zone from a long zone name that
+// includes the full path (i.e. projects/{project-id}/managedZones/{managed-zone-name}). If the data
+// cannot be parsed or does not include this information, the default (provided) values will be returned
+func ParseZoneName(providedZoneName string) (string, string, error) {
+
+	switch splitZoneName := strings.Split(providedZoneName, "/"); {
+	case len(splitZoneName) == 1:
+		return "", providedZoneName, nil
+	case len(splitZoneName) >= 4:
+		return splitZoneName[1], splitZoneName[3], nil
```

Comment:
What happens if the name is "fu/project-id/bar/managed-zone-name"? I recommend a slight tweak to the logic here...
```
case splitZoneName[0] == "projects" && splitZoneName[2] == "managedZones"
	return splitZoneName[1], splitZoneName[3], nil
```